lib/omedis/accounts/checks/can_delete_account.ex

```diff
@@ -7,13 +7,38 @@ defmodule Omedis.Accounts.CanDeleteAccount do
 
   import Ash.Query
 
+  alias Omedis.Accounts.Group
+  alias Omedis.Accounts.GroupMembership
   alias Omedis.Accounts.Organisation
 
   def describe(_options), do: "User can delete their own account"
 
   def match?(nil, _context, _options), do: false
 
   def match?(actor, _context, _options) do
-    Ash.exists?(filter(Organisation, owner_id == ^actor.id), authorize?: false)
+    Ash.exists?(
+      filter(Organisation, owner_id == ^actor.id),
+      authorize?: false
+    ) and check_admin_group_membership(actor)
```

Comment:
How about splitting this into two functions with names that immediately give you insight into what they do?
```suggestion
    owns_organisation?(actor) and is_not_the_only_admin?(actor)
```
or something more comprehensive?